subprojects/build-init/src/main/java/org/gradle/api/tasks/wrapper/Wrapper.java

```diff
@@ -427,4 +432,30 @@ public PathBase getArchiveBase() {
     public void setArchiveBase(PathBase archiveBase) {
         this.archiveBase = archiveBase;
     }
+
+    /**
+     * The network timeout specifies how many ms to wait for when the wrapper is performing network operations, such
+     * as downloading the wrapper jar.
+     *
+     * @since 7.3
+     */
+    @Option(option = "network-timeout", description = "Timeout in ms to use when the wrapper is performing network operations")
+    @Incubating
+    public void setNetworkTimeout(@Nullable String networkTimeout) {
+        this.networkTimeout = networkTimeout == null ? null : Integer.parseInt(networkTimeout);
+    }
+
+    /**
+     * The network timeout specifies how many ms to wait for when the wrapper is performing network operations, such
+     * as downloading the wrapper jar.
+     *
+     * @since 7.3
+     */
+    @Nullable
+    @Optional
+    @Input
+    @Incubating
+    public String getNetworkTimeout() {
+        return networkTimeout == null ? null : String.valueOf(networkTimeout);
+    }
```

Comment:
We don't want to introduce more non-Provider API properties on tasks, so this needs to be a `Property<Integer>`. I think our `@Option` knows how to convert String -> Integer, but if it doesn't we can add that.

```suggestion
    @Internal
    @Incubating
    @Option(option = "network-timeout", description = "Timeout in ms to use when the wrapper is performing network operations")
    public Property<Integer> getNetworkTimeout() {
        return networkTimeout;
    }
```

`networkTimeout` would need to be initialized with:

```
private final Property<Integer> networkTimeout = getProject().getObjects().property(Integer.class);
```

In addition, in `WrapperPlugin`, we should set the conventional value for this near here: https://github.com/gradle/gradle/blob/2d0225d7406122d7e345f90607438e2ddd85cc07/subprojects/build-init/src/main/java/org/gradle/buildinit/plugins/WrapperPlugin.java#L34

```
wrapper.getNetworkTimeout().convention(10000);
```